acceptance/components/pack_executor.go

```diff
@@ -0,0 +1,392 @@
+// +build acceptance
+
+package components
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"sync"
+	"testing"
+	"text/template"
+
+	"github.com/buildpacks/pack/acceptance/variables"
+
+	"github.com/buildpacks/pack/acceptance/assertions"
+
+	"github.com/Masterminds/semver"
+
+	h "github.com/buildpacks/pack/testhelpers"
+)
+
+type PackExecutor struct {
+	testObject     *testing.T
+	path           string
+	fixturesPaths  []string
+	home           string
+	registryConfig *h.TestRegistryConfig
+	assert         assertions.AssertionManager
+}
+
+//nolint:whitespace // A leading line of whitespace is left after a method declaration with multi-line arguments
+func NewPackExecutor(
+	testObject *testing.T,
+	path string,
+	fixtures []string,
+	home string,
+	registryConfig *h.TestRegistryConfig,
+	assert assertions.AssertionManager,
+) *PackExecutor {
+
+	testObject.Helper()
+
+	return &PackExecutor{
+		testObject:     testObject,
+		path:           path,
+		fixturesPaths:  fixtures,
+		home:           home,
+		registryConfig: registryConfig,
+		assert:         assert,
+	}
+}
+
+func (e *PackExecutor) Cleanup() {
+	e.testObject.Helper()
+
+	err := os.RemoveAll(e.home)
+	e.assert.Nil(err)
+}
+
+func (e *PackExecutor) cmd(name string, args ...string) *exec.Cmd {
+	e.testObject.Helper()
+
+	cmdArgs := append([]string{name}, args...)
+	cmdArgs = append(cmdArgs, "--no-color")
+	if e.Supports("--verbose") {
+		cmdArgs = append(cmdArgs, "--verbose")
+	}
+
+	cmd := e.baseCmd(cmdArgs...)
+
+	cmd.Env = append(os.Environ(), "DOCKER_CONFIG="+e.registryConfig.DockerConfigDir)
+	if e.home != "" {
+		cmd.Env = append(cmd.Env, "PACK_HOME="+e.home)
+	}
+
+	return cmd
+}
+
+func (e *PackExecutor) baseCmd(parts ...string) *exec.Cmd {
+	return exec.Command(e.path, parts...)
+}
+
+func (e *PackExecutor) RunWithCombinedOutput(name string, args ...string) (string, error) {
+	e.testObject.Helper()
+
+	output, err := e.cmd(name, args...).CombinedOutput()
+
+	return string(output), err
+}
+
+func (e *PackExecutor) SuccessfulRunWithOutput(name string, args ...string) string {
```

Comment:
I know that this is a common pattern, but when I first read it, I thought that the args provided were what we were expecting the output to be, and not the args to pack, so I was a bit confused.

Reply:
What might have made it easier to understand the intent and keep the brevity?

The interface this provides gives us 
```
output := pack.SuccessfulRunWithOutput("build", "app-name", "-B", "builder", ...)
```

from the perspective of the tests consuming the interface most things I can think of become longer
```
output := pack.CommandWithArgs("build", "app-name", "-B", "builder", ...).RunSuccessfullyReturningOutput()
```
would be more explicit.

Would renaming this to `SuccessfulRunReturningOutput()` be clearer? It adds another few characters to an already long command name.

Reply:
Hmm. I know the `...WithOutput` is a pattern, but I would find it clearer to just have `SuccesfulRun("build"...)` just return the output by default, and someone who's not interested could just `_ =` it.